src/html-serializer.js

```diff
@@ -0,0 +1,188 @@
+/*
+ * For information see
+ * https://prismic.io/docs/javascript/beyond-the-api/html-serializer
+ */
+import { escapeHtml } from './utils/escape-html'
+import { SpanNode } from './utils/nodes'
+import { resolveUrl } from './utils/resolve-url'
+import { NODE_TYPES } from './utils/types'
+
+export class HtmlSerializer {
+  /**
+   * @param {Object} [options={}] - Options for customizing the serialization
+   * @param {string} [options.imageCopyrightAttribute=data-copyright] - Attribute to assign the image copyright to
+   */
+  constructor({ imageCopyrightAttribute } = {}) {
+    this.imageCopyrightAttribute = imageCopyrightAttribute || 'data-copyright'
+  }
+
+  /**
+   * Serialize node tree into array of HTML strings.
+   *
+   * @param {NodeTree} nodeTree
+   * @returns {string[]} serialized node tree
+   */
+  serializeNodeTree(nodeTree) {
+    const step = node => {
+      const serializedChildren = node.children.reduce(
+        (result, node) => result.concat([step(node)]),
+        []
+      )
+
+      return this.serialize(node, serializedChildren)
+    }
+
+    return nodeTree.children.map(parentNode => {
+      return step(parentNode)
+    })
+  }
+
+  /**
+   * Create <tag> with children inside.
+   *
+   * @param {string} tag - Outer HTML element tag
+   * @param {string[]} children - Children concatenated and used as inner HTML
+   * @returns {string} serialized HTML string (Tag with children)
+   */
+  serializeStandardTag(tag, children) {
+    return `<${tag}>${children.join('')}</${tag}>`
+  }
+
+  /**
+   * Create <img> wrapped in <p>.
```

Comment:
An `img` is always wrapped in `p`?

Reply:
Yes, this was adopted from the original parser